stockflux-securities-master/src/components/ConfirmationButton.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react";
+import "./ConfirmationButton.css";
+import Button from "./Button";
+
+const ConfirmationButton = ({
```

Comment:
I think the approach you're making here is valid but it's not what I was looking for. Right now, you're effectively allowing the developer to use the `ConfirmationButton` component in place of the `Button`. There are some issues in your approach however, particularly in the way that you've passed the `props` down into both the button component as well as the yes button, but not the no button. You've also duplicated styles quite a bit. If the yes and no buttons both have the same style, then you can just make them use the same class.

My idea however, was to make it so that you could wrap a `Button` in a `Confirmation` component which would effectively hi-jack the click handler of that button and add the confirmation functionality. Then clicking the "yes" button would execute the button's original functionality.

```javascript
<Confirmation>
  <Button onClick={() => doSomething()} text="Button text" />
</Confirmation>
```

It might be worth looking into this approach since it will give you a deeper knowledge of how you can use React and how it works.

Reply:
K, I'll have a go at this

Reply:
Alright this has now been done with the help of Jack.